src/main/java/de/tum/in/www1/artemis/service/metis/ConversationService.java

```diff
@@ -0,0 +1,164 @@
+
+    private static final String CONVERSATION_DETAILS_ENTITY_NAME = "messages.conversationParticipant";
+
+    private static final String METIS_WEBSOCKET_CHANNEL_PREFIX = "/topic/metis/";
+
+    private final UserRepository userRepository;
+
+    private final CourseRepository courseRepository;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    private final ConversationParticipantRepository conversationParticipantRepository;
+
+    private final ConversationRepository conversationRepository;
+
+    private final SimpMessageSendingOperations messagingTemplate;
+
+    public ConversationService(UserRepository userRepository, CourseRepository courseRepository, AuthorizationCheckService authorizationCheckService,
+            ConversationParticipantRepository conversationParticipantRepository, ConversationRepository conversationRepository, SimpMessageSendingOperations messagingTemplate) {
+        this.userRepository = userRepository;
+        this.courseRepository = courseRepository;
+        this.authorizationCheckService = authorizationCheckService;
+        this.conversationParticipantRepository = conversationParticipantRepository;
+        this.conversationRepository = conversationRepository;
+        this.messagingTemplate = messagingTemplate;
+    }
+
+    /**
+     * Persists given conversation
+     *
+     * @param courseId      id of course the conversation belongs to
+     * @param conversation  conversation to be persisted
+     * @return              persisted conversation
+     */
+    public Conversation createConversation(Long courseId, Conversation conversation) {
+        final User user = this.userRepository.getUserWithGroupsAndAuthorities();
+
+        if (conversation.getId() != null) {
+            throw new BadRequestAlertException("A new conversation cannot already have an ID", CONVERSATION_ENTITY_NAME, "idexists");
+        }
+
+        if (conversation.getConversationParticipants().isEmpty()
+                || conversation.getConversationParticipants().stream().anyMatch(conversationParticipant -> conversationParticipant.getUser().getId().equals(user.getId()))) {
+            throw new BadRequestAlertException("A new conversation must have other conversationParticipants", CONVERSATION_DETAILS_ENTITY_NAME, "NotNull");
+        }
+
+        final Course course = checkUserAndCourse(user, courseId);
+        conversation.setCourse(course);
+
+        Conversation savedConversation = conversationRepository.save(conversation);
+
+        ConversationParticipant conversationParticipantOfCurrentUser = new ConversationParticipant();
+        conversationParticipantOfCurrentUser.setUser(user);
+        conversation.getConversationParticipants().add(conversationParticipantOfCurrentUser);
+
+        conversation.getConversationParticipants().forEach(conversationParticipant -> createConversationParticipant(conversationParticipant, savedConversation));
```

Comment:
For every conversationParticipant, a DB call is performed. This could be quite expensive, e.g. for large courses with 2000+ participants. Could it be an option to create all (without persistent) the conversationParticipants first and then execute a single `saveAll` call to persist them?

Reply:
In the current stage, there are only 2 participants for a conversation. In the future, there will be more for allowing group conversations. To be honest, I do not feel comfortable setting up a limit here myself. Do you have a suggestion on a number?

(I have implemented saveAll)

Reply:
Discussed it with Stephan, he does not want a limit on the number of participants.